packages/components/src/Filters/FilterDropdown.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { Dropdown, DropdownToggle, Level, DropdownProps } from '@patternfly/react-core';
+import FilterInput from './FilterInput';
+import './filter-dropdown.scss';
+
+export interface filterCategoriesProps {
+  filterCategories: [
+    {
+      title: string;
+      type: string;
+      urlParam: string;
+      values: [
+        {
+          label: React.ReactNode;
+          value: string;
+        }
+      ];
+    }
+  ];
+}
+
+export interface FilterDropdownProps extends Omit<DropdownProps, 'onSelect'> {
+  addFilter?: (filterName?: string, selectedValue?: string | number, type?: string) => void;
+  removeFilter?: (filterName?: string, selectedValue?: string | number) => void;
+  hideCategories?: [string];
+  isOpen: boolean;
+  onToggle: (isOpen: boolean) => void;
+  filters?: Record<string, string>;
+  filterCategories: filterCategoriesProps['filterCategories'];
+  onSelect: ((event?: React.SyntheticEvent<HTMLDivElement, Event> | undefined) => void) | undefined;
+  label: string;
+  ouiaId: string;
+  ouiaSafe: boolean;
+}
+
+const FilterDropdown: React.FunctionComponent<FilterDropdownProps> = ({
+  hideCategories,
+  filters,
+  filterCategories,
+  onSelect,
+  label,
+  ouiaId,
+  ouiaSafe,
+  addFilter,
+  removeFilter,
+  isOpen = false,
+  onToggle,
+  ...props
+}) => {
+  const ouiaStateId = 'RHI/FilterDropdown';
```

Comment:
If this is never re-assigned we might as well bump it out of the function. There is no point in creating this variable on each render.